parlai/tasks/genderation_bias/agents.py

```diff
@@ -0,0 +1,327 @@
+
+    def _setup_data(self, opt: Opt) -> List[List[Message]]:
+        """
+        Flatten and classify the normal task data.
+
+        Save/load where applicable.
+
+        :param opt:
+            options dict.
+        """
+        # create save directory, if it does not already exist
+        self.original_task_name = ':'.join(opt['task'].split(':')[2:])
+        self.save_dir = self._get_save_path(
+            opt['datapath'], str(datetime.datetime.today())
+        )
+        os.makedirs(self.save_dir, exist_ok=True)
+
+        fname = f"{opt['datatype'].split(':')[0]}_fourclass_{self.four_class}.json"
+        self.save_path = os.path.join(self.save_dir, fname)
+
+        data = self.load_data(opt, fname)
+        if data is not None:
+            # successfully load data
+            return data
+
+        # build the original teacher
+        original_task_module = get_original_task_module(opt)
+        teacher_opt = deepcopy(opt)
+        teacher_opt['task'] = self.original_task_name
+        teacher = original_task_module(teacher_opt)
+
+        total_exs = teacher.num_examples()
+        if self.opt['max_examples'] > 0:
+            total_exs = min(self.opt['max_examples'], total_exs)
+
+        progress_bar = tqdm(
+            total=total_exs, unit='ex', unit_scale=True, desc='Building flattened data'
+        )
+
+        all_episodes = []
+        num_exs = 0
+        while num_exs < total_exs:
+            current_episode = []
+            episode_done = False
+
+            while not episode_done:
+                action = Message(teacher.act())
+                current_episode.append(action)
+                episode_done = action.get('episode_done', False)
+                num_exs += 1
+
+            # flatten the episode into 1-example episodes with context
+            flattened_ep = flatten_and_classify(
+                current_episode,
+                opt['flatten_max_context_length'],
+                include_labels=opt['flatten_include_labels'],
+                delimiter=opt['flatten_delimiter'],
+                four_class=self.four_class,
+                word_lists=self.word_lists,
+            )
+            all_episodes += flattened_ep
+
+            progress_bar.update(len(flattened_ep))
+
+        # save data for future use
+        self.save_data(all_episodes)
+
+        return all_episodes
+
+    def load_data(self, opt: Opt, filename: str) -> Optional[List[List[Message]]]:
+        """
+        Attempt to load pre-build data.
+
+        Checks for the most recently build data via the date string.
+
+        :param opt:
+            options dict
+        :param filename:
+            name of (potentially) saved data
+
+        :return episodes:
+            return list of episodes, if available
+        """
+        # first check for the most recent date
+        save_dir = self._get_save_path(opt['datapath'], '*')
+        all_dates = []
+        for fle in glob.glob(os.path.join(save_dir, filename)):
```

Comment:
nit: `file`?

Reply:
`file` is a reserved name in Python but yeah `fle` is weird will change